tensorzero-internal/src/endpoints/datasets.rs

```diff
@@ -737,6 +737,121 @@ pub async fn delete_datapoint(
     Ok(())
 }
 
+#[derive(Debug, Deserialize)]
+pub struct ListDatapointsQueryParams {
+    limit: Option<u32>,
+    offset: Option<u32>,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct ListDatapointsPathParams {
+    dataset_name: String,
+}
+
+#[axum::debug_handler(state = AppStateData)]
+pub async fn list_datapoints_handler(
+    State(app_state): AppState,
+    Path(path_params): Path<ListDatapointsPathParams>,
+    Query(query_params): Query<ListDatapointsQueryParams>,
+) -> Result<Json<Vec<Datapoint>>, Error> {
+    list_datapoints(
+        path_params.dataset_name,
+        &app_state.clickhouse_connection_info,
+        query_params.limit,
+        query_params.offset,
+    )
+    .await
+    .map(Json)
+}
+
+#[tracing::instrument(name = "list_datapoints", skip(clickhouse))]
+pub async fn list_datapoints(
```

Comment:
We shouldn't return datapoints that are deleted/stale by default. Later we might want to add a flag to include those.

Reply:
I don't think we do?